PIM 3

NikolajBS opened the pull request.

Noob-max committed:
added interface

IProductIndex has now been created

NikolajBS committed:
Added class

NikolajBS committed:
Main to test methods

impostermp committed:
JSON file

Created JSONArray

NikolajBS committed:
temp

impostermp committed:
1

f

impostermp committed:
1

f

impostermp committed:
test

JulieSonnichsen committed:
Created test class

NikolajBS committed:
Changed interface method and added index method in ProductIndex

NikolajBS committed:
Merge remote-tracking branch 'origin/Team2.3-ProductIndex-' into Team2.3-ProductIndex-

NikolajBS committed:
Changed access modifier

AlexBirkwald committed:
Merge branch 'main' into Team2.3-ProductIndex-

NikolajBS committed:
changed import to use the correct  class

NikolajBS committed:
Updated class with methods and attributes for us to use

NikolajBS committed:
set up test

NikolajBS committed:
Added update and delete method

impostermp committed:
Update ProductIndex.java

impostermp committed:
CRUD method update

updated create/read methods, changed ean datatype from int -> long in producthit

AlexBirkwald committed:
Fixed initialization of ProductIndex in Main Method

AlexBirkwald committed:
Removed statement calling searchProducts method in ProductIndex class

AlexBirkwald committed:
Cleaned up Main Class, removing redundant code

AlexBirkwald committed:
Update ProductIndex.java

AlexBirkwald committed:
Refactored indexProductsByToken algorithm, added test to findHits method and created extra constructor in ProductHit for testing purposes

AlexBirkwald committed:
Added UI element for GUI

AlexBirkwald committed:
Merge branch 'main' into Team2.3-ProductIndex-

JulieSonnichsen committed:
Made GUIgle

JulieSonnichsen committed:
Changed GUI again

AlexBirkwald committed:
Reconstructed Interface settup.

AlexBirkwald committed:
Merge branch 'main' into Team2.3-ProductIndex-

AlexBirkwald committed:
Comitted all changes from Main branch and fixed all errors.
this includes:

- Refactor naming convention for Product (formerly ProductHit)
- Added int hitNum to Product + get/set methods
- Added implementation of Comparable interface to Product
- Implemented CompareTo method in Product.java

AlexBirkwald committed:
Added constructor in Product.java, for testing purposes

JulieSonnichsen committed:
GUI 99% finnished

MagnusHJensen commented on src/main/java/dk/sdu/se_f22/productmodule/index/ProductIndex.java:
```diff
@@ -0,0 +1,229 @@
+    private int descriptionHits = 0;
+    private List<Product> sortedList = new ArrayList<>();
+
+    private String url = "localhost";
+    private int port = 5432;
+    private String databaseName = "xxxx";
+    private String username = "xxxx";
+    private String password = "xxxx";
+    private Connection connection = null;
```
I would assume the shared DB connection should be used?

MagnusHJensen commented on src/main/java/dk/sdu/se_f22/sortingmodule/range/rangepublic/RangeFilterFilterResults.java:
```diff
@@ -20,11 +20,11 @@ public class RangeFilterFilterResults {
      */
     public static SearchHits filterResults(SearchHits searchHits, List<RangeFilter> rangeFilters) throws IllegalImplementationException {
-        Collection productHits = searchHits.getProducts();
+        Collection Products = searchHits.getProducts();
```
variable name change? Also does not follow common naming scheme.

thor2304 replied:
We (group 4.4) would appreciate that the variable is not changed. 
It is not a ClassReference and thus does not have an impact on your module.

marc7636 replied:
think they ran some project wide search and replace without being smart about it

impostermp committed:
update

updated create/read methods

NikolajBS committed:
Updated update and delete method

NikolajBS committed:
splitted index method to two seperate methods

NikolajBS committed:
Cleaned up

impostermp committed:
sql

added sql file, removed connection to local db

thor2304 reviewed (changes requested):
Please do not change our code when not necessary

impostermp committed:
123

123

impostermp committed:
Update ProductIndex.java

impostermp committed:
Update ProductIndex.java

impostermp committed:
Update 2.1_productIndex.sql

NikolajBS committed:
Changed connection to DBconnection

NikolajBS committed:
Merge remote-tracking branch 'origin/Team2.3-ProductIndex-' into Team2.3-ProductIndex-

impostermp committed:
Update ProductIndex.java

Storing searched products, as well as retrieving these products in descending order(new searches shown first)

impostermp committed:
duplicate code

removed duplicate code and put it into a separate method.

NikolajBS committed:
Fixed update method

impostermp committed:
Update ProductIndex.java

impostermp committed:
Merge branch 'Team2.3-ProductIndex-' of https://github.com/LangeHenrik/SDU-SE-F22 into Team2.3-ProductIndex-

emma7636 reviewed (changes requested):
Some small things.
Also, please pull the latest from main to fix your merge errors

Best regards
Group 2.1

marc7636 commented on src/main/java/dk/sdu/se_f22/searchmodule/infrastructure/GUI/SEM1customerGUIController.java:
```diff
@@ -65,7 +65,7 @@ public Label getBrandHitsCounter() {
-    public Label getProductHitsCounter() {
-        return productHitsCounter;
+    public Label getProductsCounter() {
+        return ProductsCounter;
```
and this

marc7636 commented on src/main/java/dk/sdu/se_f22/sortingmodule/range/rangepublic/RangeFilterFilterResults.java:
```diff
@@ -20,11 +20,11 @@ public class RangeFilterFilterResults {
         for (RangeFilter rangeFilter : rangeFilters) {
             if (rangeFilter instanceof RangeFilterClass filterCasted) {
-                productHits = filterCasted.useFilter(productHits);
+                Products = filterCasted.useFilter(Products);
```
and this

marc7636 commented on src/main/resources/dk/sdu/se_f22/searchmodule/infrastructure/GUI/SEM1customerGUI.fxml:
```diff
@@ -19,5 +19,5 @@
    <Label layoutX="215.0" layoutY="104.0" text="Product hits: " />
    <Label fx:id="brandHitsCounter" layoutX="181.0" layoutY="104.0" prefHeight="17.0" prefWidth="19.0" />
-   <Label fx:id="productHitsCounter" layoutX="286.0" layoutY="104.0" prefHeight="17.0" prefWidth="19.0" />
+   <Label fx:id="ProductsCounter" layoutX="286.0" layoutY="104.0" prefHeight="17.0" prefWidth="19.0" />
```
and this

marc7636 commented on src/test/java/dk/sdu/se_f22/searchmodule/infrastructure/GUI/SEM1customerGUIControllerTest.java:
```diff
@@ -76,7 +76,7 @@ void onSem1SearchButton(FxRobot robot) {
 
         assertEquals("1", controller.getBrandHitsCounter().getText());
-        assertEquals("0", controller.getProductHitsCounter().getText());
+        assertEquals("0", controller.getProductsCounter().getText());
```
undo all your changes in here

emma7636 reviewed (changes requested):
please undo this

NikolajBS committed:
Index products are now saved in a table

impostermp committed:
SQL code added

added additional sql code

AlexBirkwald committed:
Wired up GUI

NikolajBS committed:
small fix
